[IMP] default_warehouse_from_sale_team: Default journal for vendor bills i#17892

luisg123v opened the pull request:
Currently, when creating a customer invoice, a default journal is set
if it's configured in the sales team. However, that feature only works
for customer invoices but not for vendor bills.

This change implements the default journal feature also for vendor
bills.

In addition, now company and currency are also considered when
computing default journal, for more accurate results.

### Dummy MR:

- [vauxoo/typ!950](https://git.vauxoo.com/vauxoo/typ/-/merge_requests/950)

luisg123v commented:
Hi @desdelinux & @keylor2906,

Could you review, please?

CC @hugho-ad

hugho-ad commented on default_warehouse_from_sale_team/models/account_move.py:
```diff
@@ -7,20 +7,18 @@ class AccountMove(models.Model):
-        journal_on_team = None
-        if team and journal_types == ["sale"]:
-            journal_on_team = team._get_default_journal_sale()
+        team = self.env.context.get("salesteam", self.team_id)
```
```suggestion
        team = self.env.context.get("salesteam") or self.team_id
```

luisg123v replied:
Done.

hugho-ad commented on default_warehouse_from_sale_team/models/account_move.py:
```diff
@@ -7,20 +7,18 @@ class AccountMove(models.Model):
-        if team and journal_types == ["sale"]:
-            journal_on_team = team._get_default_journal_sale()
+        team = self.env.context.get("salesteam", self.team_id)
+        journal_on_team = team._get_default_journal(journal_types)
```
you could use 

is_sale_document/ is_purchase_document

what happen with moves created manually 

and credit notes?

luisg123v replied:
- Credit notes should also be considered
- In case of journal entries, it won't be triggered by the onchange because the onchange validates it's an invoice `is_invoice` which is `is_sale_document` + `is_purchase_document`, so the only time this could be called is when created, when `self.team_id` is not set. I could validate it, though.

luisg123v replied:
FYI I double-checked this doesn't affect journal entries and I can confirm it does not, so resolving.

CarmenMiranda commented on default_warehouse_from_sale_team/models/account_move.py:
```diff
@@ -7,20 +7,18 @@ class AccountMove(models.Model):
+            "default_currency_id": self.currency_id.id,
+        }
+        default_journal = self.with_context(**default_journal_ctx)._get_default_journal()
+        self.journal_id = default_journal
```
Hi, @luisg123v In here the `if default_journal ` is not needed anymore?

luisg123v replied:
@CarmenMiranda no, because I'm calling the native method to retrieve the default journal, which tries to find a journal based on the salesteam as first choice, but fallbacks on Odoo's native behavior. So, we know in advance there will be a journal, unless there's no journal at all.

CarmenMiranda replied:
Ohh, ok, I get it

CarmenMiranda reviewed (approved):
Technically LGTM 👍🏼

luisg123v committed:
[IMP] default_warehouse_from_sale_team: Default journal for vendor bills

Currently, when creating a customer invoice, a default journal is set
if it's configured in the sales team. However, that feature only works
for customer invoices but not for vendor bills.

This change implements the default journal feature also for vendor
bills.

In addition, now company and currency are also considered when
computing default journal, for more accurate results.

desdelinux reviewed (approved):
LGTM ready for merged @luisg123v @hugho-ad
